uwblueprint/social-diversity-for-children
# feat(translation): add locale link +  translate

### Notion ticket link



[Support Locale Options for Backend APIs](https://www.notion.so/uwblueprintexecs/Support-Locale-Options-for-Backend-APIs-04ad8fd1292148838a42d3da5496f22f)
[Add Initial batch of translations for all supported languages](https://www.notion.so/uwblueprintexecs/Support-Locale-Options-for-Backend-APIs-04ad8fd1292148838a42d3da5496f22f)





https://user-images.githubusercontent.com/44826218/136678795-a59428b8-b28b-4301-8eb6-9dcff7f22c37.mp4


### Implementation description

- Remove locale options for API backends and use client side alternative (util helpers that filter translations based on current locale, this helps us save calls to db when the user changes local)
- Given the user locale, we know get the main translation based on the locale (on a call to a class/program, we grab ALL translations) and fallback to english if the translation for the current locale does not exists
- Add translations that we received from client
- Used i18n to do translations for class, program-details, and home page
- Enable date range and weekday translations by `toLocaleDateString()`
- Fix hrefs to use `next/link`, this preserves the locale when moving from pages to pages
- Add seed for translations



### Steps to test

1. Add translations to DB (class and program translations)
2. Create an accounts (volunteer/parent)
3. Test out translations in home page via the language modal
4. Test out translations in my classes page via the language modal
5. Test out translations in program details page via language modal

In all cases, translations should show up as expected. Note that we do not have all the translations we need, we will apply the form translations later when finalizing forms. 



### What should reviewers focus on?

- Dev: Kinda sketchy implementation of `weekdayToString` 👀 
- Dev: Link replacements `next/link` wrappers
- Design: Look and feel of translation/language modal flow
- Design: maybe try out the deployed instance!

### Checklist
